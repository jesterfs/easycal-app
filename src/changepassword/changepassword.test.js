import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import ChangePassword from './changepassword'
import renderer from 'react-test-renderer';

describe('ChangePassword component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <BrowserRouter>
        <ChangePassword />
      </BrowserRouter>,
      div
    )
    ReactDOM.unmountComponentAtNode(div)
  })
  
  it('renders the UI as expected', () => {
    const div = document.createElement('div')
    const tree = renderer
      .create(<BrowserRouter>
        <ChangePassword />
      </BrowserRouter>,
      div)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });

});