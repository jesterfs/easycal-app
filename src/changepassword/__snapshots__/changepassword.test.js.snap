// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangePassword component renders the UI as expected 1`] = `
<div
  className="changepassword, greetgroup"
>
  <div
    className="item"
  >
    <h2>
      Change Your Password
    </h2>
  </div>
  <div
    className="item"
  >
    <form
      className="change-password-form"
      onSubmit={[Function]}
    >
      <div>
        <label
          htmlFor="newpassword"
        >
          New Password
        </label>
        <input
          id="newpassword"
          name="newpassword"
          type="password"
        />
      </div>
      <div>
        <label
          htmlFor="newpasswordagain"
        >
          Confirm Password
        </label>
        <input
          id="newpassword2"
          name="newpassword2"
          type="password"
        />
      </div>
      <button
        type="submit"
      >
        Change Password
      </button>
    </form>
  </div>
</div>
`;
